src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -128,7 +128,13 @@ void OTARequestor::OnQueryImageResponse(void * context, const QueryImageResponse
             ChipLogDetail(SoftwareUpdate, "Update available from %" PRIu32 " to %" PRIu32 " version",
                           requestorCore->mCurrentVersion, update.softwareVersion);
             MutableByteSpan updateToken(requestorCore->mUpdateTokenBuffer);
-            CopySpanToMutableSpan(update.updateToken, updateToken);
+            // This function copies the bytespan to mutablebytespan only if size of mutablebytespan buffer is greater or equal to
+            // bytespan otherwise we are copying data upto available size.
+            err = CopySpanToMutableSpan(update.updateToken, updateToken);
+            if (err == CHIP_ERROR_BUFFER_TOO_SMALL)
+            {
+                memcpy(updateToken.data(), update.updateToken.data(), updateToken.size());
```

Comment:
So why is it ok to drop the part of the update token that does not fit?

Reply:
I emptied buffer and gererated updateToken. In specs it is mentioned that use nodeId if we don’t  have token.

Reply:
I don't think that's the right approach. We should only use the nodeId as token if we've lost the token. In this case, the token is invalid and it feels like we should treat it as such. We should error out here or in the original location (where you removed in this PR).

@tcarmelveilleux thoughts?

Reply:
@carol-apple This is a outdated code. Pl check here https://github.com/project-chip/connectedhomeip/commit/50a2383577dafe62a958184f4deec2bdc744cde9#:~:text=requestorCore%2D%3EGenerateUpdateToken()%3B